### PR TITLE
docs(developer): kmc-ldml api documentation

### PR DESCRIPTION
**Pull request by mcdurdin**
Relates to #10207.

Note: reduces the API surface by removing members which should be internal. Fixes a few minor typos in other API docs. Adds missing docs for kmc-kmn also.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-ldml/src/compiler/compiler.ts [hunk 1]
```diff
@@ -41,29 +41,74 @@ export const SECTION_COMPILERS = [
...
 
+  /**
+   * Initialize the compiler, including loading the WASM host for uset parsing.
+   * Copies options.
+   * @param callbacks - Callbacks for external interfaces, including message
+   *                    reporting and file io
+   * @param options   - Compiler options
+   * @returns           false if initialization fails
+   */
   async init(callbacks: CompilerCallbacks, options: LdmlCompilerOptions): Promise<boolean> {
```
Comment (rc-swag) [inline developer/src/kmc-ldml/src/compiler/compiler.ts:95]: are these auto-generated? Noticing it is inserthing a hyphen after the param names and before the description but no hyphen on the @returns. Wondering if it is an intential change.

**Review by rc-swag — APPROVED** [review]
lgtm 

Just a query about the introduction of hyphens in the comment blocks.

File: developer/src/kmc-ldml/src/compiler/compiler.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/src/kmc-ldml/src/compiler/compiler.ts:95]: The hyphen is optional for jsdoc and required for tsdoc, it seems.

> The @param tag is followed by a parameter name, followed by a hyphen, followed by a description.

https://tsdoc.org/pages/tags/param/

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.282-beta](https://keyman.com/downloads/releases/beta/17.0.282)

**Pull request merged**